checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -223,6 +253,10 @@ protected void refineGT(
             knownToBeLessThanLengthOf(array, right, store, in);
         }
 
+        if (right instanceof NumericalAdditionNode || right instanceof NumericalSubtractionNode) {
+            propagateToSubterms((BinaryOperationNode) right, refinedRight, store);
```

Comment:
So I don't think it makes sense for them to share code, since it would require the code to become quite complicated with special cases.